src/mjolnir/landmark_builder.cc

```diff
@@ -104,28 +157,21 @@ LandmarkDatabase::LandmarkDatabase(const std::string& db_name, bool read_only)
     : pimpl(new db_pimpl(db_name, read_only)) {
 }
 
-void LandmarkDatabase::insert_landmark(const Landmark& landmark) {
+void LandmarkDatabase::insert_landmark(const std::string& name,
+                                       const LandmarkType& type,
+                                       const double lng,
+                                       const double lat) {
   auto* insert_stmt = pimpl->insert_stmt;
   if (!insert_stmt)
     throw std::logic_error("Sqlite database connection is read-only");
 
   sqlite3_reset(insert_stmt);
   sqlite3_clear_bindings(insert_stmt);
 
-  if (landmark.name != "") {
-    sqlite3_bind_text(insert_stmt, 1, landmark.name.c_str(), landmark.name.length(), SQLITE_STATIC);
-  } else {
-    sqlite3_bind_null(insert_stmt, 1);
-  }
-
-  if (landmark.type != "") {
-    sqlite3_bind_text(insert_stmt, 2, landmark.type.c_str(), landmark.type.length(), SQLITE_STATIC);
-  } else {
-    sqlite3_bind_null(insert_stmt, 2);
-  }
-
-  sqlite3_bind_double(insert_stmt, 3, landmark.lng);
-  sqlite3_bind_double(insert_stmt, 4, landmark.lat);
+  sqlite3_bind_text(insert_stmt, 1, name.c_str(), name.length(), SQLITE_STATIC);
+  sqlite3_bind_int(insert_stmt, 2, static_cast<int>(type));
+  sqlite3_bind_double(insert_stmt, 3, lng);
+  sqlite3_bind_double(insert_stmt, 4, lat);
```

Comment:
I removed the empty checks for name and type here because in common use cases `insert_landmark` should only be called by `node_callback` where checks and initializations already happen.